src/gas-price/gas-price.ts

```diff
@@ -102,6 +102,38 @@ export const calculateScalingMultiplier = (
     maxScalingMultiplier
   );
 
+export const fetchAndStoreGasPrice = async (
+  chainId: string,
+  providerName: string,
+  provider: ethers.JsonRpcProvider
+) => {
+  // Get the provider recommended gas price and save it to the state
+  logger.debug('Fetching gas price and saving it to the state.');
+  const goGasPrice = await go(async () => {
+    const feeData = await provider.getFeeData();
+    // We assume the legacy gas price will always exist. See:
+    // https://api3workspace.slack.com/archives/C05TQPT7PNJ/p1699098552350519
+    return feeData.gasPrice;
+  });
+  const gasPrice = goGasPrice.data;
+  if (!goGasPrice.success) {
+    if (!goGasPrice.success) logger.error('Failed to fetch gas price from RPC provider.', goGasPrice.error);
+    return null;
+  }
```

Comment:
This seems like a typo:
```suggestion
  if (!goGasPrice.success) {
    logger.error('Failed to fetch gas price from RPC provider.', goGasPrice.error);
    return null;
  }
```

Reply:
Yeah, thanks.